lib/os/json.c

```diff
@@ -509,12 +509,18 @@ static int arr_parse(struct json_obj *obj,
 {
 	ptrdiff_t elem_size = get_elem_size(elem_descr);
 	void *last_elem = (char *)field + elem_size * max_elements;
-	size_t *elements = (size_t *)((char *)val + elem_descr->offset);
+	size_t *elements = NULL;
 	struct token value;
 
+	if (val) {
+		elements = (size_t *)((char *)val + elem_descr->offset);
+	}
+
 	__ASSERT_NO_MSG(elem_size > 0);
 
-	*elements = 0;
+	if (elements) {
```

Comment:
IMO it's more natural to check the pointer you're about to write to for NULL instead of some other variable.
Also, the hope is that it prevents bugs caused by future changes, e.g. if element doesn't rely on `var` anymore.